sources/packages/api/src-sql/DeleteCascades/Drop-cascade-rules.sql

```diff
@@ -0,0 +1,339 @@
+-- Delete reference between applications and student_assessments. --
+ALTER TABLE
+    sims.applications DROP CONSTRAINT IF EXISTS applications_current_assessment_id_fkey;
+
+-- Create reference between application_exception_requests and application_exceptions without delete cascade. --
+ALTER TABLE
+    sims.applications
+ADD
+    CONSTRAINT applications_current_assessment_id_fkey FOREIGN KEY(current_assessment_id) REFERENCES sims.student_assessments(id);
+
+-- Delete reference between application_exception_requests and application_exceptions. --
+ALTER TABLE
+    sims.application_exception_requests DROP CONSTRAINT IF EXISTS application_exception_requests_application_exception_id_fkey;
+
+-- Create reference between application_exception_requests and application_exceptions without delete cascade. --
+ALTER TABLE
+    sims.application_exception_requests
+ADD
+    CONSTRAINT application_exception_requests_application_exception_id_fkey FOREIGN KEY(application_exception_id) REFERENCES sims.application_exceptions(id);
+
+-- Delete reference between student_assessments and applications. --
+ALTER TABLE
+    sims.student_assessments DROP CONSTRAINT IF EXISTS student_assessments_application_id_fkey;
+
+-- Create reference between student_assessments and applications without delete cascade. --
+ALTER TABLE
+    sims.student_assessments
+ADD
+    CONSTRAINT student_assessments_application_id_fkey FOREIGN KEY(application_id) REFERENCES sims.applications(id);
+
+-- Delete reference between applications and students. --
+ALTER TABLE
+    sims.applications DROP CONSTRAINT IF EXISTS applications_student_id_fkey;
+
+-- Create reference between applications and students without delete cascade. --
+ALTER TABLE
+    sims.applications
+ADD
+    CONSTRAINT applications_student_id_fkey FOREIGN KEY(student_id) REFERENCES sims.students(id);
+
+-- Delete reference between application_student_files and applications. --
+ALTER TABLE
+    sims.application_student_files DROP CONSTRAINT IF EXISTS application_student_files_student_file_id_fkey;
+
+-- Create reference between application_student_files and applications without delete cascade.--
+ALTER TABLE
+    sims.application_student_files
+ADD
+    CONSTRAINT application_student_files_student_file_id_fkey FOREIGN KEY(application_id) REFERENCES sims.applications(id);
```

Comment:
<img width="722" alt="image" src="https://user-images.githubusercontent.com/62901416/182492885-0bb501ae-cb08-4aee-8d95-067041464403.png">

I guess there is a confusion here, the constraint name is application_student_files_student_file_id_fkey but the reference is application id. Also one cascade is missing.

Reply:
Fixed